### ISSUE TITLE
ui/ux: tree node styling issue in the `references` view

### ISSUE THREAD

**Issue by vladarama** [eclipse-theia/theia#12796]
### Bug Description:
In the `References` view, the styling of the tree nodes flickers when changing node. The flickering seems to be caused by the `focused` styling not being applied correctly.

Bug Demo:
![references styling bug](https://github.com/eclipse-theia/theia/assets/86936229/eb791856-0d4d-43a0-979c-5fdfb366df63)

Correct Styling Demo (`Explorer View`):
![correct reference styling](https://github.com/eclipse-theia/theia/assets/86936229/cfa513f7-f231-4cb5-bea8-ac31a6800752)

### Steps to Reproduce:

1. Start Theia application
2. Select a random text in your editor -> Right-Click -> `Find All References` (`Alt+Shift+F12`)
3. Go to the `References` view
4. Click on different results
5. Notice the incorrect styling of the nodes

### Additional Information

- Operating System: Windows 10
- Theia Version: 1.40

**Comment by shynxe** [comment]
I'm trying to look into this. As I understand, this `References` view comes with theia's `vscode-builtin-extensions`. Since the vscode's extensions source code can not be modified, then it's Theia's API that should be looked into, correct?

**Comment by vince-fugnitto** [comment]
> ...then it's Theia's API that should be looked into, correct?

That's correct, there is likely something incorrect with how we handle selection in trees contributed by the plugin-system.

**Comment by shynxe** [comment]
Okay, after looking into this for a bit, I discovered this: when a `LineNode` is selected, the line is focused in the editor and it is due to this behavior that the `TreeWidget` loses focus, and therefore loses its styling. Clicking anywhere on the `TreeView` will refocus it and reapply the styling, resulting in the _flickering_ mentioned in the issue.
 
![chrome-capture-2023-7-3](https://github.com/eclipse-theia/theia/assets/60783301/676159d9-688e-4366-a5e4-3feb9141d89a)

In VS Code's implementation of this `LineNode` selection, it seems that the editor is not focused when the line is brought into view. Should I look into this behavior and make it so that the `TreeWidget` maintains focus?
